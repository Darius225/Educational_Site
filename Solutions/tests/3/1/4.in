500
179 415 254 446 157 928 771 609 788 786 12 136 279 860 603 903 446 704 79 769 460 978 345 775 919 621 218 692 949 951 188 200 666 936 598 298 271 575 747 108 989 150 333 162 962 152 437 995 979 655 735 116 491 674 612 241 859 455 600 45 883 261 430 603 815 614 256 622 205 67 347 38 534 793 578 596 316 210 250 849 299 103 782 490 770 572 836 926 108 156 79 234 762 687 934 397 267 113 357 136 223 965 387 217 472 66 956 632 722 4 488 80 981 134 844 978 761 543 287 207 33 43 279 973 18 428 525 836 838 711 22 297 916 988 131 43 723 561 580 428 127 502 817 657 82 150 377 638 879 692 454 44 643 516 496 404 805 380 937 601 751 234 505 65 133 552 292 36 382 950 856 417 193 526 579 436 458 398 901 564 404 209 41 517 121 519 554 945 832 135 660 956 335 282 896 368 209 1 796 374 185 217 562 998 144 134 640 857 444 135 840 53 801 824 200 593 756 309 645 746 606 910 628 285 861 301 114 405 577 72 788 815 278 891 537 77 229 774 238 16 248 159 610 673 862 121 300 362 11 221 257 464 580 919 459 395 474 981 142 810 621 732 861 473 442 812 220 292 952 124 671 360 719 656 763 960 634 378 691 262 759 64 15 954 77 5 287 376 725 626 102 557 284 341 589 882 104 594 117 252 376 722 842 923 577 514 452 953 190 805 407 854 744 796 837 882 135 469 394 970 343 82 409 751 534 724 998 395 94 471 331 373 518 103 478 927 21 235 891 662 394 74 694 49 117 387 12 110 50 777 579 810 624 743 311 72 597 96 790 483 884 914 566 503 489 196 917 877 820 870 129 901 597 247 18 649 207 581 893 239 679 320 322 880 111 776 424 167 486 137 191 276 123 476 590 993 658 856 14 468 917 8 47 151 943 318 58 572 445 504 662 141 670 187 551 747 685 198 428 653 476 920 104 484 331 788 3 701 881 403 73 446 665 322 892 932 369 975 941 92 672 758 105 656 622 393 714 173 633 335 835 187 397 748 880 940 435 490 773 985 573 781 614 513 824 493 207 731 299 832 781 367 903 791 605 128 232 882 635 196 652 526 757 87 801 549 625 364 235 302 932 174 406 399 837 118 257 817 674 310 
250
616
103
269
42
695
493
-27
768
815
658
326
181
745
84
38
-82
-67
615
486
-40
377
157
477
0
745
28
595
105
-95
728
632
527
225
-77
381
716
776
138
-5
897
316
722
243
376
321
171
887
207
897
467
199
119
674
598
838
462
742
829
486
130
183
812
392
583
825
361
241
275
749
473
-86
431
99
469
831
153
199
494
346
456
216
484
112
287
285
877
444
302
808
428
27
741
312
896
496
810
-63
511
57
821
191
891
442
-38
419
788
776
799
636
101
856
767
306
843
121
430
408
195
424
558
480
-53
747
401
19
221
183
216
105
632
717
115
621
863
891
50
324
29
751
-45
600
657
690
445
145
647
410
797
811
527
504
543
781
666
460
463
441
802
525
704
432
-95
94
501
896
405
869
454
352
-62
645
685
394
65
718
834
194
696
772
60
743
697
710
537
528
96
895
167
121
503
194
755
510
212
506
675
583
613
352
-25
364
491
560
637
267
845
182
478
176
414
53
242
556
538
432
555
329
140
-3
569
476
196
694
50
-44
592
722
-12
355
824
148
149
179
822
50
811
338
294
81
-40
601
458
478
489
9
-39
664
655
279
633
