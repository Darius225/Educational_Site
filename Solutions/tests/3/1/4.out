-1
217
124
333
-1
399
-1
-1
-1
-1
998
209
-1
770
747
-1
-1
-1
625
-1
581
380
882
179
-1
949
-1
66
-1
-1
-1
-1
301
-1
320
-1
-1
580
-1
-1
135
-1
673
207
82
417
-1
857
-1
731
374
207
-1
-1
-1
614
-1
-1
625
22
517
-1
123
-1
-1
914
159
960
-1
791
-1
446
136
832
-1
516
374
837
12
435
756
801
981
376
5
-1
622
842
-1
881
692
-1
744
-1
257
-1
-1
-1
455
-1
956
-1
105
-1
653
-1
-1
-1
-1
965
-1
-1
452
-1
43
73
445
368
331
-1
652
-1
-1
8
769
910
517
756
66
-1
-1
978
-1
-1
-1
735
534
951
-1
-1
-1
-1
-1
393
150
-1
662
-1
-1
-1
-1
-1
-1
-1
573
513
758
-1
-1
-1
665
-1
934
-1
-1
318
-1
880
624
-1
-1
-1
590
614
-1
-1
896
-1
-1
883
-1
-1
-1
-1
-1
267
-1
36
43
-1
896
-1
-1
801
-1
-1
-1
-1
624
-1
489
174
-1
-1
292
-1
41
635
458
551
674
610
-1
-1
665
-1
471
127
-1
-1
232
209
-1
735
-1
-1
-1
-1
72
-1
879
692
564
-1
735
-1
891
589
103
-1
-1
773
635
302
786
-1
-1
-1
262
-1
